### DISCUSSION TITLE
sortie rmd

### DISCUSSION THREAD

**Discussion by gdauby** [dreamRs/conrappli#20] [General]
Voici un bout de code pour l'exportation des résultats par espèce. 

L'idéal serait de retrouver dans cet onglet la possibilité de sélectionner une espèce et donc d'exporter un rapport pour cette espèce.

```
### exportation d'un rapport espèce par espèce
if (isTruthy(input$taxa) && !identical(input$taxa, "All")) {
  
  data <- dplyr::filter(data, .__taxa == input$taxa)
  
  ### pas sur que ce soit ça pour appeler les données spatiales validées
  data_sf <- reactive(rv$spatial_data) %>% dplyr::filter(data, .__taxa == input$taxa)
  
  tax <- input$taxa
  
  rmarkdown::render(
  'conrapli_report.Rmd',
  params = list(
    tax = tax,
    data = data, ## 
    data_sf = data_sf,
    res_aoo = aoo_res$AOO_poly %>% filter(tax == !!tax), ### sortie du module critère B - rv$aoo_res
    res_eoo = eoo_res$spatial %>% filter(tax == !!tax), ### sortie du module critère B - rv$eoo_res
    res_loc = locations$locations_poly %>% filter(tax == !!tax), ### sortie du module critère B - rv$locations
    threat_sig = spatial_data, ### sortie du module mapping - rv$spatial_data
    parameters = parameters, ### nouvelle sortie du module critère B - rv$results
    results = results ### sortie du module critère B - rv$results
  ),
  output_file = paste0("report_conr_", gsub(" ", "_", tax), '.html')
)
  
}


```

**Comment by gdauby** [comment]


**Reply by gdauby** [reply]
Voici le modèle Rmd pour commencer

**Comment by gdauby** [comment]
@pvictor est ce que tu as pu voir les proposition ci dessus pour implémenter la sortie ?

**Comment by gdauby** [comment]
@pvictor je suis en train de mettre à jour la sortie Rmd en locale, je vais pusher la nouvelle version directement sur la branche principale

**Reply by pvictor** [reply]
Hello Gilles, ça me parait bien tout ça, je m'en occupe dans la semaine !

**Reply by gdauby** [reply]
https://github.com/dreamRs/conrappli/commit/8d1699789bb7dfc3985218a4f9412f01f41c0dc7

**Comment by gdauby** [comment]
Il y aura deux sorties Rmd : 

- La première (ci dessus) est une sortie pour une espèce sélectionnée. 

- La deuxième est un rapport pour toutes les espèces. Je viens de pusher une première version de ce deuxième Rmd.
https://github.com/dreamRs/conrappli/commit/122dc7b1a2958203600600c0c624e07eb746fd9e

Pour ce Rmd, les paramètres d'entrée sont les sorties d'analyse du critère B et les données importées, y inclus le shapefile 
(soit importée `shp_r <- data_shapefile_server(id = "shp")`, soit créé).

**Reply by pvictor** [reply]
A quoi correspond le paramètre `res_loc` dans le rapport species_report.Rmd ?

**Reply by gdauby** [reply]
C'est la sortie `locations` de `criterion_b`

**Comment by gdauby** [comment]
donc si la sortie dans de `conr-server.R` est telle que je l'ai proposé, ca devrait être quelque chose comme 
`criterion_b$locations$locations_poly %>% filter(tax == !!tax)`, filtré pour avoir l'espèce sélectionné. J'espère que c'est clair.

**Reply by pvictor** [reply]
Ok, mon problème est avec ces filtres : https://github.com/dreamRs/conrappli/blob/94e1a388493472a560d051ac7fad12829202a086/inst/reports/species_report.Rmd#L130
Je ne trouve pas la variable `threat`

**Comment by gdauby** [comment]
Je vérifie. Est ce que tu as installé la dernière version @devel de ConR ?

**Reply by pvictor** [reply]
ah c'est p-e ça

**Reply by gdauby** [reply]
En fait c'est parce que cette variable n'existe pas si aucun shapefile n'est utilisé dans la fonction de ConR, donc c'est ma faute, je dois modifier de mon côté.

**Reply by gdauby** [reply]
Je peux modifier le Rmd ?

**Reply by pvictor** [reply]
ok d'ac, yes tu peux modifier

**Reply by gdauby** [reply]
Voila ca devrait être bon.

Pour la sortie du rapport, je sais pas si c'était clair ci dessus, mais l'idée serait d'avoir un menu déroulant sur les espèces identiques à celui qu'on voit dans le module-critrerion-b ou l'utilisateur peut choisir une espèce ou toutes les espèces. Et selon l'un ou l'autre, c'est l'un ou l'autre rapport qui est généré.

**Reply by pvictor** [reply]
good :+1:

**Reply by pvictor** [reply]
ok! je comprend, je fait ça
